src/tests/test_queueing.py

```diff
@@ -0,0 +1,209 @@
+            "c": [],
+            "a": [],
+        }
+
+        serder = serdering.SerderKERI(sad=icp, kind=eventing.Serials.json)
+        msg = serder.raw
+
+        queue = queueing.Queueing(hab=hab)
+        queue.push_to_queued(hab.pre, msg)
+
+        # Verify keldb_queued had events
+        keldb_queued_items = [(pre, serder) for (pre,), serder in queue.keldb_queued.getItemIter()]
+        assert keldb_queued_items != []
+
+        queue.publish()
+
+        # Verify keldb_queued published and remove from keldb_queued
+        keldb_queued_items = [ (pre, serder) for (pre, ), serder in queue.keldb_queued.getItemIter()]
+        assert keldb_queued_items == []
+
+        # Verify event published and keldb_published had events from keldb_queued
+        keldb_published = [(pre, serder) for (pre,), serder in queue.keldb_published.getItemIter()]
+        assert keldb_published != []
+
+
+def test_http_call():
+    salter = Salter(raw=b'0123456789abcdef')
+    vsn = vesn = 0  # sn and last establishment sn = esn
+    salter = Salter(raw=b'0123456789abcdef')
+    with openDB(name="edy") as db, openKS(name="edy") as ks, habbing.openHby(name="keria", salt=salter.qb64, temp=True) as hby:
+        # Init key pair manager
+        mgr = Manager(ks=ks, salt=salter.qb64)
+        hab = hby.makeHab("test02")
+
+        mgr = Manager(ks=ks, salt=salter.qb64)
+        verfers, digers = mgr.incept(icount=1, ncount=0, transferable=True, stem="C")
+        #
+        # # Test with inception message
+        # serder = incept(keys=[verfers[0].qb64], code=MtrDex.Blake3_256)
+
+        qry = hab.query(pre=hab.pre, src=hab.pre, route="/log")
+        serder = serdering.SerderKERI(raw=qry)
+        # Test with query message
+        ked = serder.ked
+
+        sigers = mgr.sign(ser=serder.raw, verfers=verfers)  # default indexed True
+        assert isinstance(sigers[0], Siger)
+        msg = messagize(serder, sigers=sigers)
+
+        srdr = serdering.SerderKERI(raw=qry)
+
+        qserder = query(route="log",
+                        query=dict(i='DAvCLRr5luWmp7keDvDuLP0kIqcyBYq79b3Dho1QvrjI'),
+                        stamp=help.helping.DTS_BASE_0)
+
+        msg = qserder.raw
+
+        CESR_CONTENT_TYPE = "application/cesr+json"
+        CESR_ATTACHMENT_HEADER = "CESR-ATTACHMENT"
+        CESR_DESTINATION_HEADER = "CESR-DESTINATION"
+
+        attachment = b'-VAj-HABEIaGMMWJFPmtXznY1IIiKDIrg-vIyge6mBl2QV8dDjI3-AABAAB6P97k'
+
+        headers = Hict(
+            [
+                ("Content-Type", CESR_CONTENT_TYPE),
+                ("Content-Length", str(len(msg))),
+                (CESR_ATTACHMENT_HEADER, attachment),
+                (
+                    CESR_DESTINATION_HEADER,
+                    "BD27gP7-sD8XSr7_tTo54aNIRzPBJGX7GvRUVojfYL2H",
+                ),
+            ]
+        )
+
+        res = requests.request(
+            "POST", "http://localhost:5666/", headers=headers, data=msg
+        )
+        assert res.status_code == 204
```

Comment:
I'm not sure what this is testing. My question was if you send something to the API that isn't a KEL, if it will fully succeed and try to call `eventing.loadEvent` and then fail there.

Reply:
for the HTTP post method, cardano-backer accept the query method (ked["t"] == "qry" only, the order method will return the stauts 204 (if success)

We do have some fail scenarios
it not cershttprequest
=> then the return status will be 400, since we've already have the validation when parse https://github.com/cardano-foundation/cardano-backer/blob/feat/watchpoc/src/backer/backering.py#L299

For the KEL,the serdering.SerderKERI will perform validate the payload
=> If the KEL is not valid, or st not KEL, they will raise exception with status code return is 400, https://github.com/cardano-foundation/cardano-backer/blob/feat/watchpoc/src/backer/backering.py#L300

Reply:
But @minhdo6491 `on_post` calls `self.rxbs.extend(msg)` which puts it on the queue to be parsed and then returns 204. It kicks off a `QryRpyMailboxIterable` -> `MailboxIterable` in the background too.

It's on the parsing queue in the background so it could still fail, and now in `MailboxIterable` it might not be able to load the event, right?

Quick question: how does the integration from KERIA to the backer work when creating an identifier? It sends a `qry` message...? Or it sends a `icp` message and then `qry` message after or something. We need to understand this in detail

Reply:
`on_post calls self.rxbs.extend(msg) which puts it on the queue to be parsed and then returns 204. It kicks off a QryRpyMailboxIterable -> MailboxIterable in the background too.`
From my local test (both integration and UTs)), it seems not kick the QryRpyMailboxIterable -> MailboxIterable in the background

for the integration flow
`Quick question: how does the integration from KERIA to the backer work when creating an identifier? It sends a qry message...? Or it sends a icp message and then qry message after or something. We need to understand this in detail`
From my understand, the icp will send first then qry will come after

Reply:
hmm so it returns 204 and does not kick off `QryRpyMailboxIterable -> MailboxIterable`? that's not the right response then. Why doesn't it kick off `QryRpyMailboxIterable -> MailboxIterable`?

Reply:
well, I can help debug it
i may took a while